File: Python version/print_r.py
'''
- print_r()
- string print_r (mixed object [, bool view = False])
- Prints human-readable information about the object
- @author: Alexander Guinness
- @version: 1.0
- @params: {mixed} The Object to be printed
- @params: {view} Optional boolean parameter to set an alternative view
- @return string represented by standard output function print()
- @license: MIT
- @date: 2/27/12 9:28 PM
'''
__all__ = ["print_r"]

def print_r(obj, view = 0):
	def build(obj, indent = ''):
		data = []

		if isinstance(obj, list) and not indent:
			for key, value in enumerate(obj):
				data.extend(['[', key, '] => ', value, '\n'])
		else:
			def depth(item):
				def get_view(i):
					return ''.join(view and ['\t[', i, '] => '] or ['\t', i, ': '])

				block = ['{\n']

				for key, value in sorted(item):
					block.extend([indent, get_view(key), build(value, indent + '\t'), ',', '\n'])

				block.pop(-2)
				data.extend([''.join(block), indent, '}'])

			if isinstance(obj, dict):
				depth(obj.items())

			elif '__dict__' in dir(obj) and type(obj) is not type:
				depth(obj.__dict__.items())

			elif isinstance(obj, str):
				data.extend(['"', obj, '"'])

			else:
				data.append(obj)

		return ''.join(map(str, data))
	print(build(obj))

File: JavaScript version/print_r.js
/**
 * print_r()
 * string print_r (mixed object [, boolean view = false])
 * Prints human-readable information about the object
 * @author: Alexander Guinness
 * @version: 1.0
 * @params: {mixed} The Object to be printed
 * @params: {view} Optional boolean parameter to set an alternative view
 * @return String
 * @license: MIT
 * @date: 2/27/12 9:28 PM
**/

function print_r(object, view) {
	return function self(object, indent) {
		var type = Object.prototype.toString.call(object),
			data = [];

		if (type === '[object Array]' && !indent) {
			var i = object.length;

			while (i--)
				data.unshift('[', i, '] => ', object[i], '\n');
		}
		else {
			if (type === '[object Object]') {
				indent = indent || '';

				var get_view = function(i) {
					return (view ? ['\t[', i, '] => '] : ['\t', i, ': ']).join('');
				},
				block = ['{\n'];

				for (var i in object)
					block.push(indent, get_view(i), self(object[i], indent + '\t'), ',', '\n');

				block.splice(-2, 1);
				data.push(block.join(''), indent, '}');
			}
			else {
				data.push({
					'[object String]': '"' + object + '"',
					'[object Array]' : '[' + object + ']'
				}[type] || object);
			}
		}
		return data.join('');
	}(object);
}

//exports.toCSS